plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/button/frontend.ts

```diff
@@ -51,14 +55,59 @@ const { state: addToCartWithOptionsState } = store< AddToCartWithOptionsStore >(
 	{ lock: universalLock }
 );
 
+const getMatchedCartItem = (
+	cartItem: OptimisticCartItem,
+	selectedItem: CartVariationItem[]
+) => {
+	if (
+		! Array.isArray( cartItem.variation ) ||
+		! Array.isArray( selectedItem )
+	)
+		return false;
+	if ( cartItem.variation.length !== selectedItem.length ) return false;
+
+	return cartItem.variation.every(
+		( {
+			raw_attribute,
+			value,
+		}: {
+			raw_attribute: string;
+			value: string;
+		} ) =>
+			selectedItem.some( ( item ) => {
+				const selectedItemAttr = item.attribute;
+				const selectedItemValue = item.value.toLowerCase();
```

Comment:
Nitpick: I would remove these assignments and instead use `item.attribute` and `item.value` directly in the return below. I think it will make it easier to understand.